apps/modernization-ui/src/SkipLink/SkipLinkContext.tsx

```diff
@@ -22,16 +24,37 @@ interface SkipLinkProviderProps {
 }
 
 export const SkipLinkProvider = ({ children }: SkipLinkProviderProps) => {
-    const [id, skipTo] = useState<string | undefined>(undefined);
+    const [currentFocusTarget, setCurrentFocusTarget] = useState('');
+    const [isActionFocused, setIsActionFocused] = useState(true);
+
+    const skipTo = (id: string) => {
+        setCurrentFocusTarget(id);
+        setIsActionFocused(true);
+    };
+
+    const focusedTo = (isFocused: boolean) => {
+        setIsActionFocused(isFocused);
+    };
 
     // Setting up the id to pass it to the anchor tag
     const contextValue: SkipLinkContextType = {
-        skipTo
+        skipTo,
+        focusedTo,
+        currentFocusTarget
     };
+
+    useEffect(() => {
+        const targetElement = document.getElementById(currentFocusTarget);
+        if (targetElement && isActionFocused) {
+            targetElement.focus();
+            setCurrentFocusTarget('');
```

Comment:
question:  Should the focus target be cleared at this point?  Does it need to remain as the main content?

Reply:
yes we need this because if we remove it then after the search is made if the user makes another search then the focus would stay at the last Name and not go the results count because the skipTo is already assigned to the results count so we need to clear it after the focus is made